test/functional/pdo_sqlsrv/pdostatement_GetDataType.phpt

```diff
@@ -1,305 +1,302 @@
 
-function testsmalldatetime($db)
+function testDatetimeoffset($db, $tbname)
 {
-	$stmt = $db->query("SELECT SmallDTCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT DTOffsetCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testDateTime($db)
+function testsmalldatetime($db, $tbname)
 {
-	$stmt = $db->query("SELECT DateTimeCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT SmallDTCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testDate($db)
+function testDateTime($db, $tbname)
 {
-	$stmt = $db->query("SELECT DateCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT DateTimeCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testNVarcharMax($db)
+function testDate($db, $tbname)
 {
-	$stmt = $db->query("SELECT NVarCharMaxCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT DateCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testTime($db)
+function testNVarcharMax($db, $tbname)
 {
-	$stmt = $db->query("SELECT TimeCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT NVarCharMaxCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testUniqueidentifier($db)
+function testTime($db, $tbname)
 {
-	$stmt = $db->query("SELECT Guidcol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT TimeCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testVarbinaryMax($db)
+function testUniqueidentifier($db, $tbname)
 {
-	$stmt = $db->query("SELECT VarbinaryMaxCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT Guidcol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testVarcharMax($db)
+function testVarbinaryMax($db, $tbname)
 {
-	$stmt = $db->query("SELECT VarcharMaxCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT VarbinaryMaxCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
 
-function testXml($db)
+function testVarcharMax($db, $tbname)
 {
-	$stmt = $db->query("SELECT XmlCol FROM PDO_AllTypes");
-	$result = $stmt->fetch(PDO::FETCH_ASSOC);
-	var_dump($result);
+    $stmt = $db->query("SELECT VarcharMaxCol FROM $tbname");
+    $result = $stmt->fetch(PDO::FETCH_ASSOC);
+    var_dump($result);
 }
-	
 
```

Comment:
Please refactor this test by having only one method that also takes the column name. Most of the above methods are the same